init-tracing-opentelemetry/src/tracing_subscriber_ext.rs

```diff
@@ -88,22 +88,34 @@ where
     let layer = tracing_opentelemetry::layer()
         .with_error_records_to_exceptions(true)
         .with_tracer(tracerprovider.tracer(""));
-    global::set_tracer_provider(tracerprovider);
-    Ok(layer)
+    global::set_tracer_provider(tracerprovider.clone());
+    Ok((layer, TracingGuard { tracerprovider }))
 }
 
-pub fn init_subscribers() -> Result<(), Error> {
+pub struct TracingGuard {
+    tracerprovider: trace::TracerProvider,
+}
+
+impl Drop for TracingGuard {
+    fn drop(&mut self) {
+        self.tracerprovider.force_flush();
+    }
+}
+
+pub fn init_subscribers() -> Result<TracingGuard, Error> {
```

Comment:
Suggestion annotate with `#[must_use]` to raise a warning when the output is not managed.

If it doesn't behave as expected, I'll deprecate the previous signature as suggested by @nshalman

Reply:
I had to put the `#[must_use]` on the struct instead of the function, as on the function it acts on the `Result` instead of the `TracingGuard`.

It seems to behave well to me - it's a warning instead of an error, so existing code will still compile without any changes.